lib/chef/resource/chocolatey_installer.rb

```diff
@@ -0,0 +1,198 @@
+class Chef
+  class Resource
+    class ChocolateyInstaller < Chef::Resource
+      provides :chocolatey_installer
+
+      description "Use the Chocolatey Installer resource to ensure that Choco is installed to your specification. Use the Chocolatey Feature resource to customize your install"
+      introduced "18.3"
+      examples <<~DOC
+          **Install Chocolatey**
+
+          ```ruby
+          chocolatey_installer 'latest' do
+            action :install
+          end
+          ```
+
+          **Uninstall Chocolatey**
+
+          ```ruby
+          chocolatey_installer 'Some random verbiage' do
+            action :uninstall
+          end
+          ```
+
+          **Install Chocolatey with Parameters**
+
+          ```ruby
+          chocolatey_installer 'latest' do
+            action :install
+            download_url "https://www.contoso.com/foo"
+            chocolatey_version '2.12.24'
+          end
+          ```
+
+          **Upgrade Chocolatey with Parameters**
+
+          ```ruby
+          chocolatey_installer 'latest' do
+            action :upgrade
+            chocolatey_version '2.12.24'
+          end
+          ```
+      DOC
+
+      allowed_actions :install, :uninstall, :upgrade
+
+      property :download_url, String,
+        description: "The URL to download Chocolatey from. This defaults to the value of $env:ChocolateyDownloadUrl, if it is set, and otherwise falls back to the official Chocolatey community repository to download the Chocolatey package. It can be used for offline installation by providing a path to a Chocolatey.nupkg."
+
+      property :chocolatey_version, String,
+        description: "Specifies a target version of Chocolatey to install. By default, the latest stable version is installed. This will use the value in $env:ChocolateyVersion by default, if that environment variable is present. This parameter is ignored if download_url is set."
```

Comment:
Same

Reply:
Per our discussion, this is actually being used directly by the Chocolatey `install.ps1` script and the environment variable can be used to redirect the standard installer to an alternate _version_.